src/Tools/dotnet-stack/ReportCommand.cs

```diff
@@ -121,7 +121,11 @@ private static async Task<int> Report(CancellationToken ct, IConsole console, in
                         // Thread id is in the frame name as "Thread (<ID>)"
                         string template = "Thread (";
                         string threadFrame = stackSource.GetFrameName(stackSource.GetFrameIndex(stackIndex), false);
-                        int threadId = int.Parse(threadFrame.Substring(template.Length, threadFrame.Length - (template.Length + 1)));
+
+                        // we are looking for the first index of ) because
+                        // we need to handle a thread name like: Thread (4008) (.NET IO ThreadPool Worker)
+                        int firstIndex = threadFrame.IndexOf(")");
```

Comment:
```suggestion
                        int firstIndex = threadFrame.IndexOf(')');
```
The `char`-overload is cheaper than the `string`-ovrelaod.

Reply:
done